genologics/lims.py

```diff
@@ -692,8 +704,19 @@ def get_batch(self, instances, force=False):
             data = self.tostring(ElementTree.ElementTree(root))
             root = self.post(uri, data)
             for node in list(root):
-                instance = instance_map[node.attrib["limsid"]]
+                uri = node.attrib['uri']
+                if uri in instance_map:
+                    instance = instance_map[uri]
+                else:
+                    # We're getting a uri we didn't ask for. This should mean that we
+                    # asked for one without the state flag but are getting one with it
+                    parsed = urlparse(uri)
+                    uri_without_state_param = "{}://{}{}".format(
+                            parsed.scheme, parsed.netloc, parsed.path)
+                    instance = instance_map[uri_without_state_param]
+                    node.attrib['uri'] = uri_without_state_param
```

Comment:
Line 707 - 717 is what works for us (tested the Scilifelab code and that didn't work in this seection)